src/client.cc

```diff
@@ -207,8 +209,13 @@ int Client::InitialStep(sasl_client_params_t *params,
   if (err != SASL_OK) return err;
 
   user_ = auth_name;
-  token_ = TokenStore::Create(log_.get(), password);
-  if (!token_) return SASL_FAIL;
+
+  if (Config::Get()->external_token_manager() == false) {
+    token_ = TokenStore::Create(log_.get(), password);
+    if (!token_) return SASL_FAIL;
+  } else {
+    tokenfile_ = password;
+  }
```

Comment:
I think it's cleaner overall if instead of having two different code paths here, in `Client`, we push this complexity into `TokenStore` by adding a flag there that just disables refreshes. `Client` would remain unmodified except that it passes the value of `manage_token_externally` to `TokenStore`, and `TokenStore` would just always return whatever's in the token file, regardless of the expiry time.